File: Android/FitIn_v2/app/src/main/java/com/example/fitin_v2/ui/onboard/signup/SignUpFirstViewModel.java
package com.example.fitin_v2.ui.onboard.signup;

import android.app.Application;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.fitin_v2.model.AccountRequestDto;
import com.example.fitin_v2.model.AccountResponseDto;
import com.example.fitin_v2.repository.UserRepository;

public class SignUpFirstViewModel extends AndroidViewModel {

    public MutableLiveData<String> name = new MutableLiveData<>();
    public MutableLiveData<String> emailId = new MutableLiveData<>();
    public MutableLiveData<String> email = new MutableLiveData<>();
    public MutableLiveData<String> password = new MutableLiveData<>();

    private MutableLiveData<AccountRequestDto> accountDto;

    private final MutableLiveData<Boolean> _eventBack = new MutableLiveData<Boolean>();
    LiveData<Boolean> eventBack;


    public SignUpFirstViewModel(@NonNull Application application) {
        super(application);

    }

    public LiveData<Boolean> getBack() {
        return eventBack = _eventBack;
    }

    public void onBack() {
        _eventBack.setValue(true);
    }

    public void onBackComplete() {
        _eventBack.setValue(false);
    }

    public MutableLiveData<AccountRequestDto> getAccount() {
        if (accountDto == null) {
            accountDto = new MutableLiveData<>();
        }
        return accountDto;
    }

    public void getUser(View view) {
        String emails = emailId.getValue() + "@" + email.getValue();
        AccountRequestDto accountRequestDto = new AccountRequestDto(emails,password.getValue(),name.getValue());
        accountDto.setValue(accountRequestDto);
    }



    public LiveData<AccountRequestDto> getEventSignUpComplete() {
        return accountDto = null;
    }


}

File: Android/FitIn_Kotlin/app/src/main/java/com/example/fitin_kotlin/ui/onboard/signup/SignUpFirstViewModel.kt
package com.example.fitin_kotlin.ui.onboard.signup

import android.view.View
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.fitin_kotlin.data.model.network.request.RequestSignUp
import dagger.hilt.android.lifecycle.HiltViewModel

class SignUpFirstViewModel : ViewModel(){

    val name: MutableLiveData<String> = MutableLiveData<String>()
    val id: MutableLiveData<String> = MutableLiveData<String>()
    val email: MutableLiveData<String> = MutableLiveData<String>()
    val password: MutableLiveData<String> = MutableLiveData<String>()

    private val _requestSignUp: MutableLiveData<RequestSignUp?> = MutableLiveData<RequestSignUp?>()
    val requestSignUp: LiveData<RequestSignUp?>
        get() = _requestSignUp

    fun getRequestSignUp(view: View) {
        val email: String = id.value + "@" + email.value
        val signUp = RequestSignUp(email, password.value, name.value, "010231321313")
        _requestSignUp.value = signUp
    }

    fun getRequestSignUpComplete() {
        _requestSignUp.value = null
    }

    private val _eventBack = MutableLiveData<Boolean>()
    val eventBack: LiveData<Boolean>
        get() = _eventBack

    fun onBack() {
        _eventBack.value = true
    }

    fun onBackComplete() {
        _eventBack.value = false
    }

}